docs/en/routing/basic-usage.md

```diff
@@ -0,0 +1,490 @@
+const routerWithHistoryOverride = new Router(routes, { HistoryManager: StateHistory });
+```
+
+Or using the `registerRouterInjector` helper function:
+
+>src/main.ts
+```ts
+import Registry from '@dojo/framework/widget-core/Registry';
+import { registerRouterInjector } from '@dojo/framework/routing/RouterInjector';
+import StateHistory from '@dojo/framework/routing/history/StateHistory';
+
+import routes from './routes';
+
+const registry = new Registry();
+
+// creates and registers a router using the default history manager, `HashHistory`
+registerRouterInjector(routes, registry);
+
+// creates and registers a router using the `StateHistory`
+registerRouterInjector(routes, registry, { HistoryManager: StateHistory });
+```
+
+### HashHistory
+
+ `HashHistory` uses the fragment identifier to process route changes, for example `https://foo.com/#home` would process the `home` as the route path. As `HashHistory` is the default manager, you do not need to import module.
+
+ ```ts
+import { Router } from '@dojo/framework/routing/Router';
+
+const router = new Router(config);
+```
+
+ ### StateHistory
+
+`StateHistory` uses the browser's [history API](https://developer.mozilla.org/en-US/docs/Web/API/History), to manage application route changes.
+
+**Note:** The `StateHistory` manager will require server-side machinery to enable an application to support refreshing on a route.
+
+```ts
+import { Router } from '@dojo/framework/routing/Router';
+import { MemoryHistory } from '@dojo/framework/routing/history/StateHistory';
+
+const router = new Router(config, { HistoryManager: StateHistory });
+```
+
+ ### MemoryHistory
+
+The `MemoryHistory` does not rely on any browser API but keeps its own internal path state. It should not be used in production applications but is useful for testing application routing.
+
+```ts
+import { Router } from '@dojo/framework/routing/Router';
+import { MemoryHistory } from '@dojo/framework/routing/history/MemoryHistory';
+
+const router = new Router(config, { HistoryManager: MemoryHistory });
+```
+
+>src/main.tsx
+```tsx
+import renderer from '@dojo/framework/widget-core/vdom';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import { registerRouterInjector } from '@dojo/framework/routing/RouterInjector';
+
+import routes from './routes';
+import App from './App';
+
+const registry = new Registry();
+// creates a router with the routes and registers the router with the registry
+registerRouterInjector(routes, registry);
+
+const r = renderer(() => <App />);
+r.mount({ registry });
+```
+
+These history managers work like adapters, meaning that custom history managers can be implemented by fulfilling the history manager interface.
+
+## Error Outlet
+
+A special `outlet` called, `errorOutlet` is registered for that will match when the route doesn't match (`exact` or `partial`) any outlet in the routing configuration. You can use this `outlet` to render a widget to inform the user that the route does not exist.
```

Comment:
How do error outlets and default routes interact?

Reply:
I'll add some notes, on the default route only applies to the initial application load. If there is a default route it will redirect to it, otherwise when using the application the default route doesn't have any affect.